File: product-service/src/main/java/com/intexsoft/webshop/productservice/service/SubcategoryService.java
package com.intexsoft.webshop.productservice.service;

import com.intexsoft.webshop.productservice.dto.subcategory.SubcategoryCreateDto;
import com.intexsoft.webshop.productservice.dto.subcategory.SubcategoryDto;
import com.intexsoft.webshop.productservice.dto.subcategory.SubcategoryUpdateDto;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface SubcategoryService {
    SubcategoryDto createSubcategory(SubcategoryCreateDto subcategoryCreateDto);

    SubcategoryDto findSubcategoryById(Long subcategoryId);

    List<SubcategoryDto> findSubcategories(Pageable pageable);

    SubcategoryDto updateSubcategory(Long subcategoryId, SubcategoryUpdateDto subcategoryUpdateDto);

    void deleteSubcategoryById(Long subcategoryId);
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/service/SubcategoryService.kt
package com.intexsoft.webshop.productservicekt.service

import com.intexsoft.webshop.productservicekt.dto.subcategory.SubcategoryCreateDto
import com.intexsoft.webshop.productservicekt.dto.subcategory.SubcategoryDto
import com.intexsoft.webshop.productservicekt.dto.subcategory.SubcategoryUpdateDto
import org.springframework.data.domain.Pageable

interface SubcategoryService {
    fun createSubcategory(subcategoryCreateDto: SubcategoryCreateDto): SubcategoryDto
    fun findSubcategoryById(subcategoryId: Long): SubcategoryDto
    fun findSubcategories(pageable: Pageable): List<SubcategoryDto>
    fun updateSubcategory(subcategoryId: Long, subcategoryUpdateDto: SubcategoryUpdateDto): SubcategoryDto
    fun deleteSubcategoryById(subcategoryId: Long)
}